### DISCUSSION TITLE
Semi-OTS Heatsink

### DISCUSSION THREAD

**Discussion by ayilm1** [jankae/LibreVNA#228] [Show and tell]
Hi all,

I just wanted to put up this post documenting the modification of an off-the-shelf 120x100x18mm extruded heatsink for use on the LibreVNA.

The off-the-shelf items purchased for this mod are as follows:

- 120x100x18mm aluminium extruded heatsink (have a look around ebay/amazon/etc. Should be easy to find).
- 120x120x0.3mm thermal pad. Grease works too if you don't mind using it.
- QTY 5, M3x14mm cap-head screws. Note: I ordered M3x12mm for mine after basing the stack height off the shield models on git. 12mm worked fine for the corner holes but I think the thread for the centre hole is slightly lower and 12mm was not long enough to reach it and is thus not installed on mine currently.

The stock heatsink looks like this:
![Stock Heatsink - Bottom](https://github.com/jankae/LibreVNA/assets/2467074/5209d5db-bab9-4373-9cd4-b8d668e836d0)
![Stock Heatsink - Side](https://github.com/jankae/LibreVNA/assets/2467074/211933e1-ed6f-4e69-a32c-9dbc4fc7d517)
![Stock Heatsink - Top](https://github.com/jankae/LibreVNA/assets/2467074/480ca88c-ab0f-4775-866a-bf6401b4b873)

Here's the thermal pad:
![Thermal Pad](https://github.com/jankae/LibreVNA/assets/2467074/b033a607-bbfc-4e38-8277-2deeed6a84ba)

I grabbed the hole centres off the bottom shielding can model in the git repo and marked them out on the bottom of the heatsink:
![Hole Locations](https://github.com/jankae/LibreVNA/assets/2467074/fb35277e-b781-4df7-812e-6173cd51372f)
![Hole Drilling](https://github.com/jankae/LibreVNA/assets/2467074/759672b4-3b4f-45d8-99d0-cc3bd98b73a0)

The corner holes are too close to the edge and just break out, so its best to just open these up entirely.
![Corner Hole - Bottom](https://github.com/jankae/LibreVNA/assets/2467074/3719442c-7301-4441-85ee-612fe3b70afa)
![Corner Hole - Top](https://github.com/jankae/LibreVNA/assets/2467074/4b8624cd-f38c-4f1e-9dd1-ae30e5940321)
![Drilled Holes - Bottom](https://github.com/jankae/LibreVNA/assets/2467074/6d565d47-3a17-4678-8f61-d5a1d0a20529)
![Drilled Holes - Top](https://github.com/jankae/LibreVNA/assets/2467074/d05d9f8c-0af2-4f05-8d9d-bfc1021e1708)

You will also need to shorten the end fins to allow clearance for the screw heads:
![Shortened End Fins - Top](https://github.com/jankae/LibreVNA/assets/2467074/bb8e08a9-3f66-40c1-9dca-1b9c6cde0020)
![Shortened End Fins](https://github.com/jankae/LibreVNA/assets/2467074/1eba1f1e-57d2-45a9-bc4a-b3a4d27d80a5)

Finally, I rounded off the corners of the fins to keep it from scratching things/cutting fingers:
![Fillets](https://github.com/jankae/LibreVNA/assets/2467074/68c303e4-59e7-4243-8ea1-c4968fc4717c)
![Modified Heatsink - Top](https://github.com/jankae/LibreVNA/assets/2467074/741372d6-95e2-4a5e-b738-e629c99c4c15)

We also need to cut holes into the thermal pad for the screws to pass through:
![Modified Thermal Pad](https://github.com/jankae/LibreVNA/assets/2467074/ad2245fd-6471-4066-9a68-53aec59c1c5f)

Here is the replacement screw for the original M3x8mm. Again, I recommend 14mm length in hindsight. Pictured is 12mm.
![Screws](https://github.com/jankae/LibreVNA/assets/2467074/268931b8-7657-4174-9a71-31fc279564c4)

Finally, here's how it looks installed:
![Installed 1](https://github.com/jankae/LibreVNA/assets/2467074/1cbb7dea-df73-4a52-b035-4b76e2e79c41)
![Installed 2](https://github.com/jankae/LibreVNA/assets/2467074/fc2b2e97-d7de-41d7-9d22-ddd5b6cf1c7e)

I've also attached a step model of the heatsink with the above modifications applied (provided without warranty).
[Heatsink.zip](https://github.com/jankae/LibreVNA/files/13317625/Heatsink.zip)

### Testing
I ran a couple of baseline tests first with no heatsink fitted. I did one run with the ports open and another with 50R on each to see if there was any difference in power dissipation. There wasn't (the unit started the second test with a higher temperature which explains why it reaches 60C more quickly).

Three readings were taken at each time step; the maximum PLL temperature as reported in the UI, and top and side surface temperatures as measured using a thermal camera. As can be seen, the PLL temperature seems to settle at around 60C after around 1.5 hours of operation. The ambient temperature at the start of the test was 26.2C and 27.9C at the end.
![image](https://github.com/jankae/LibreVNA/assets/2467074/f01054a0-a57b-40a6-bfe3-c6c1755cd870)
![image](https://github.com/jankae/LibreVNA/assets/2467074/a56208f0-c646-4998-aa0c-f2800c8557a6)
![VNA - Tape](https://github.com/jankae/LibreVNA/assets/2467074/6b70e677-3045-4ad2-9a3a-5d41cd92d3c5)
![image](https://github.com/jankae/LibreVNA/assets/2467074/41877a45-1977-46f6-9b50-ef473f2c7c39)

The same tests were repeated with the heatsink installed. One test was run with the heatsink fins facing down (LEDs visible) and another with them facing up. As expected, performance is better with the fins up as air isn't trapped.
![image](https://github.com/jankae/LibreVNA/assets/2467074/0573bba1-9880-4eac-9a74-7f33d6e167c5)
![image](https://github.com/jankae/LibreVNA/assets/2467074/68875dc1-6072-4615-8a44-5a6ba1e33bb8)
![image](https://github.com/jankae/LibreVNA/assets/2467074/7788eb68-f32a-4dd4-82e9-0b16bd83685c)
![image](https://github.com/jankae/LibreVNA/assets/2467074/b5ecdcb6-5e40-40f7-99dd-fa4966ccf6e7)

As can be seen, the installation of the heatsink results in a reduction in internal temperature of ~10C with the heatsink sub-optimally oriented (to permit viewing of the LEDs) and ~15C when correctly oriented. The delta between the internal and surface temperature is consequently lowered, consistent with the lower thermal impedance of the added heatsink.

**Comment by jankae** [comment]
Nice and thorough documentation, thanks for sharing this :)

**Reply by ayilm1** [reply]
No problem, thanks for the design of a quality, polished, and affordable instrument. Feel free to include the heatsink model in the repo if you think it will be of use. Might even be possible to make an alternate bottom shield with the fins integrated for a v1.1.
